pkg/client/restclient/request.go

```diff
@@ -805,10 +805,12 @@ func (r *Request) request(fn func(*http.Request, *http.Response)) error {
 		}
 		resp, err := client.Do(req)
 		updateURLMetrics(r, resp, err)
-		if err != nil {
-			r.backoffMgr.UpdateBackoff(r.URL(), err, 0)
-		} else {
-			r.backoffMgr.UpdateBackoff(r.URL(), err, resp.StatusCode)
+		if r.baseURL != nil {
```

Comment:
For now, baseUrlKey() will return the host simply, and if r.baseURL==nil, r.URL() will contain no host info, we do not need to call UpdateBackoff(). just like the Watch() and Stream() functions do in the same file.